### PR TITLE
[ADD] sale_order_back2draft: Back sale orders to draft state

### PR DESCRIPTION
**Pull request by pedrobaeza**
More info in the README.

### CODE REVIEW

**Comment by rdeheele** [comment]
:+1:

**Comment by pedrobaeza** [comment]
Can anyone review this?

File: sale_order_back2draft/models/sale_order.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    _inherit = "sale.order"
+
+    @api.multi
+    def button_draft(self):
+        # go from canceled state to draft state
+        for order in self:
+            if order.state != 'cancel':
+                raise exceptions.Warning(
+                    _("You can't back any order that it's not on cancel "
+                      "state"))
```
Comment (laetitia-gangloff) [inline sale_order_back2draft/models/sale_order.py]: Nice to have : sale order name in the warning

File: sale_order_back2draft/models/sale_order.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+            order.order_line.write({'state': 'draft'})
+            order.write({'state': 'draft'})
```
Comment (laetitia-gangloff) [inline sale_order_back2draft/models/sale_order.py:21]: Is it not possible to have only one write for the state ?

File: sale_order_back2draft/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (c) 2014 Serv. Tecnol. Avanzados (http://www.serviciosbaeza.com)
```
Comment (laetitia-gangloff) [inline sale_order_back2draft/__openerp__.py]: 2015 ?

**Comment by laetitia-gangloff** [comment]
The rest seems ok :+1:

File: sale_order_back2draft/models/sale_order.py (hunk 2 above)
Comment (pedrobaeza) [inline sale_order_back2draft/models/sale_order.py:21]: Well, this is to assure that workflow is not changed by the state value, and it's not too much performance cost (this is usually called for one record).

File: sale_order_back2draft/__openerp__.py (hunk 3 above)
Comment (pedrobaeza) [inline sale_order_back2draft/__openerp__.py]: Well, I did it last year, but now is when I publish it. I can put both years.

**Comment by pedrobaeza** [comment]
@laetitia-gangloff, fix your remarks, thanks.

**Comment by ghost** [comment]
:+1:

**Pull request merged**